src/sidebar/store/create-store.js

```diff
@@ -7,6 +7,85 @@ import immutable from '../util/immutable';
 
 import { createReducer, bindSelectors } from './util';
 
+/**
+ * Helper that strips the first argument from a function type.
+ *
+ * @template F
+ * @typedef {F extends (x: any, ...args: infer P) => infer R ? (...args: P) => R : never} OmitFirstArg
+ */
```

Comment:
Credit: https://stackoverflow.com/a/58765199/434243.

This relies on a feature known as [conditional types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-inference-in-conditional-types). To clarify what is going on here, it can be described in words as:

"If the type F is compatible with `(x: any, ...args: P) => R`, where `P` and `R` are to be deduced automatically, then return the type `(...args: P) => R`. Otherwise trigger an error."